### PR TITLE
Rewrite ReceivedItems and new game delivery

### PR DESCRIPTION
**Pull request by heinermann**
This should deal with issues #109, and #73 which could technically still happen, and a few other edge case issues. For example moving before receiving Connected would set `LOAD_KEY` and then just ignore what it was supposed to do in Connected. 

Mountain entrance items on new game are based on the receive cache instead of waiting for a `ReceivedItems` sync packet, so they should spawn in slightly quicker and avoid cases like #73 when the connect didn't happen quickly or other reasons.

Removes the need to sync items at all when connected, saving some more bandwidth.

### CODE REVIEW

**Comment by heinermann** [comment]
Needs test but lazy cat being lazy tonight.

**Comment by ScipioWright** [comment]
Potions, dice, and spell refreshes are currently redelivering with this update

**Comment by ScipioWright** [comment]
Picking up a chest containing an orb did not immediately increase my orb counter. When restarting, it redelivered the orb to me but didn't spawn anything in the starting cave. When restarting again, it correctly increased the orb counter on spawned and populated the cave.

**Comment by ScipioWright** [comment]
If you delete your cache then load into a game that has already sent checks to you, it creates issues.
- Hearts spawn on top of you on spawn instead of incrementing your health
- Only one orb spawned when there should have been 2, and it also spawned on me instead of just incrementing the orb counter
- No other items spawned (expected a wand, a sadekivi, a broken wand, and a gourd at minimum)

This is an issue since that would be the same as joining a game late (for an async, or if you just had some technical issues and couldn't start immediately).

File: init.lua [hunk 1]
```diff
@@ -251,41 +249,50 @@ function SendConnect()
 end
 
 
+local function SpawnAllNewGameItems()
+	local ng_items = {}
+	for _, item in ipairs(Cache.ItemDelivery:reference()) do
+		local item_id = item["item"]
+		ng_items[item_id] = (ng_items[item_id] or 0) + 1
+	end
```
Comment (ScipioWright) [inline init.lua:257]:
```suggestion
local function SpawnAllNewGameItems()
	local ng_items = {}
	for _, item in ipairs(Cache.ItemDelivery:reference()) do
		local item_id = item["item"]
		if item_table[item_id].newgame then
			ng_items[item_id] = (ng_items[item_id] or 0) + 1
		end
	end
```

Comment (ScipioWright) [inline init.lua:257]:
```suggestion
local function SpawnAllNewGameItems()
	local ng_items = {}
	for _, item in ipairs(Cache.ItemDelivery:reference()) do
		local item_id = item["item"]
		if item_table[item_id].newgame then
			ng_items[item_id] = (ng_items[item_id] or 0) + 1
		end
	end
```

**Comment by heinermann** [comment]
I'm guessing as far as orbs are concerned, the number needs to be the total number of orbs and not counting a single orb like it's doing?

**Comment by ScipioWright** [comment]
That's right

**Comment by ScipioWright** [comment]
[logger.txt](https://github.com/DaftBrit/NoitaArchipelago/files/11413546/logger.txt)

Re-joined the RC5 async that has one of each game in it, took a while to finish getting connected (expected), and then my items spawned all on top of me.

From reading the logger, after receiving the items, it attempted to do the spawn starting items, but didn't have any items to spawn with that. Then it did location scouts, then it spawned the rest of the items on me.

**Pull request merged**